config/next/config.serviceWorker.js

```diff
@@ -1,83 +1,121 @@
-const withOffline = require("next-offline");
+const { InjectManifest } = require("workbox-webpack-plugin");
+const crypto = require("crypto");
+const fs = require("fs");
+const path = require("path");
 const { PHASE_DEVELOPMENT_SERVER } = require("next/constants");
 
+const { srcDir } = require("./constants");
+const { exportSw } = require("./utils");
+
+const sericeWorkerFile = "service-worker.js";
+const sericeWorkerPath = `/static/${sericeWorkerFile}`;
+const serviceWorkerDest = `.next${sericeWorkerPath}`;
+const serviceWorkerUrl = `/${sericeWorkerFile}`;
+
+const generateInDevMode = isDev =>
+  isDev ? process.env.SERVICE_WORKER === "true" : true;
+
 module.exports = (nextConfig = {}, { nextComposePlugins, phase }) => {
-  const swDest = "static/service-worker.js";
-  const config = {
+  const addSw = generateInDevMode(phase === PHASE_DEVELOPMENT_SERVER);
+
+  return {
     ...nextConfig,
+
     env: {
       ...nextConfig.env,
-      SERVICE_WORKER_EXISTS: true,
-      SERVICE_WORKER_TIMEOUT: "60000",
+      ...(addSw && {
+        SERVICE_WORKER_EXISTS: true,
+        SERVICE_WORKER_TIMEOUT: "60000",
+        SERVICE_WORKER_URL: serviceWorkerUrl,
+      }),
     },
-    target: "serverless",
-    generateInDevMode: process.env.SERVICE_WORKER === "true",
-    sourcemap: true,
-    transformManifest: manifest => ["/"].concat(manifest),
-    workboxOpts: {
-      swDest,
-      exclude: [
```

Comment:
There is not need for excluding those paths?

Reply:
It's moved to l: 84